### ISSUE TITLE
Question about Time axis

### ISSUE THREAD

**Issue by jbkopecky** [VSainteuf/lightweight-temporal-attention-pytorch#1]
Hello Vivien,

Sorry to bother again, I have a quick question about time index:

Is it correct that you need to have precisely the same time sampling to score the model ? For example if the model was trained on 28 dates in 2017, every input tensor must have a sample set of pixels at each of these dates: (28, C, S) ?

Thanks a lot !

**Comment by VSainteuf** [comment]
Hi Jb,
That's a good question, the quick answer is no in theory, but because of my implementation yes.
But there are workarounds.

So the L-TAE in theory can be applied to sequences of varying length, and also be trained on sequences of different lengths than for inference. 
But in the implementation of this repo, because I had a dataset in which all time series have exactly the same length, I did not pay attention to this and there's one bad design choice, which is that the positional encodings are computed as a table once and for all when the LTAE is instantiated (line 57 of models/ltae.py). The only freedom you have is the 'len_max_seq' argument that defines the size of the positional encoding table that is computed. But when you will load the pre-trained weights with a different value for 'len_max_seq'  this will clash because the positional encoding table saved is not of the same size.

So I would recommend instantiating the LTAE with 'len_max_seq' set high enough for your data. Then in the model's weight dict, remove the key corresponding to the positional encoding (it's not learned anyway). Eventually, loading the weights with the option `strict=False` should do the trick. 

In [a later version of my code](https://github.com/VSainteuf/utae-paps/blob/main/src/backbones/positional_encoding.py) I coded the positional encoding as a module that computes the encodings on the fly. This requires the data loader to also yield the dates of each image, but makes the implementation much more flexible regarding varying lengths. You can also have a look to this and see if you can adapt it in your code.

**Comment by jbkopecky** [comment]
Merci Vivien !
